### ISSUE TITLE
Minifier results in "TypeError: Cannot convert undefined or null to object" error

### ISSUE THREAD

**Issue by lencioni** [evanw/esbuild#1767]
I am using [esbuild's minifier via esbuild-loader](https://github.com/privatenumber/esbuild-loader#js-minification-eg-terser) in some webpack config for a storybook build, and I have found that on some code it ends up producing output that has this runtime error:

```
TypeError: Cannot convert undefined or null to object
    at hasOwnProperty (<anonymous>)
    ... (unintelligible bits of stack trace)
    at i.unstable_runWithPriority (file://....)
    ... (more unintelligible bits of stack trace)
```

The `unstable_runWithPriority` I believe indicates that this runtime error was triggered during React rendering, in case that's helpful in anyway.

I've dug through some of the minified code in the stack trace, but was unable to find the `hasOwnProperty` that's listed here. One possible clue is that the function at the top of the stack trace seems to be a generator. I was able to trace it back to the original source code and it looks like this function was originally an `async` function that appears to have been compiled down to a generator by Babel.

If I disable esbuild-loader's minifier and use the default terser, this error does not happen. This also does not happen when not using a minifier at all.

I've reproduced this in 0.13.9 and 0.13.13. I haven't tried other versions.

**Comment by evanw** [comment]
Sorry but I'm not sure what to do with this bug report. There isn't enough information here to know what to fix, if anything. So I'm marking this as unactionable.

One example of information that would be helpful is a single file of code that runs correctly without being minified but that doesn't run correctly when minified with esbuild. To cut out the potential that it's some other non-esbuild bug (such as a bug with esbuild-loader), I would make sure that running the bare `esbuild --minify` command on the file causes the problem.

**Comment by lencioni** [comment]
I've done some more digging and I'm not sure if this is going to be terribly helpful but I wanted to record what I've found in case it helps someone else searching. Feel free to close out this issue.

I have discovered that the problem was triggered by configuring the esbuild-loader minifier plugin like this:

```ts
new ESBuildMinifyPlugin({
  target: 'es2015',
})
```

When I wrote this, I had assumed that the `target` option would work like terser's `ecma` option, but unexpectedly, esbuild-loader's ESBuildMinifyPlugin will actually just pass this on down to esbuild which makes it compile the code down to that target. This is what was compiling the async/await code to generators.

I fixed this all by changing the configuration to:

```ts
new ESBuildMinifyPlugin({
  minify: true,
})
```

I attempted to put together a minimal repro case by writing some simple async/await code and then building it like this:

```sh
npx esbuild --target=es2015 --minify index.js
```

This did end up compiling to generators, but executing the code did not throw the same error.

I'm not planning to dig anymore into this, so hopefully this helps a future explorer!
